scala-package/examples/src/main/java/org/apache/mxnetexamples/javaapi/infer/bert/BertQA.java

```diff
@@ -0,0 +1,146 @@
+        // Get the formatted logits result
+        NDArray startLogits = NDArray.reshape(
+                NDArray.new reshapeParam(output[0]).setShape(new Shape(new int[]{0, -3})))[0];
+        NDArray endLogits = NDArray.reshape(
+                NDArray.new reshapeParam(output[1]).setShape(new Shape(new int[]{0, -3})))[0];
+        // Get Probability distribution
+        float[] startProb = NDArray.softmax(
+                NDArray.new softmaxParam(startLogits))[0].toArray();
+        float[] endProb = NDArray.softmax(
+                NDArray.new softmaxParam(endLogits))[0].toArray();
+        int startIdx = argmax(startProb);
+        int endIdx = argmax(endProb);
+        return tokens.subList(startIdx, endIdx + 1);
+    }
+
+    public static void main(String[] args) throws Exception{
+        BertQA inst = new BertQA();
+        CmdLineParser parser = new CmdLineParser(inst);
+        parser.parseArgument(args);
+        BertUtil util = new BertUtil();
+        Context context = Context.cpu();
+        if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
+                Integer.valueOf(System.getenv("SCALA_TEST_ON_GPU")) == 1) {
+            context = Context.gpu();
+        }
+        // pre-processing - tokenize sentence
+        List<String> tokenQ = util.tokenizer(inst.inputQ.toLowerCase());
+        List<String> tokenA = util.tokenizer(inst.inputA.toLowerCase());
+        int validLength = tokenQ.size() + tokenA.size();
+        logger.info("Valid length: " + validLength);
+        // generate token types [0000...1111....0000]
+        List<Float> QAEmbedded = new ArrayList<>();
+        util.pad(QAEmbedded, 0f, tokenQ.size()).addAll(
+                util.pad(new ArrayList<Float>(), 1f, tokenA.size())
+        );
+        List<Float> tokenTypes = util.pad(QAEmbedded, 0f, inst.seqLength);
+        // make BERT pre-processing standard
+        tokenQ.add("[SEP]");
+        tokenQ.add(0, "[CLS]");
+        tokenA.add("[SEP]");
+        tokenQ.addAll(tokenA);
+        List<String> tokens = util.pad(tokenQ, "[PAD]", inst.seqLength);
+        logger.info("Pre-processed tokens: " + Arrays.toString(tokenQ.toArray()));
+        // pre-processing - token to index translation
+        util.parseJSON(inst.modelVocab);
+        List<Integer> indexes = util.token2idx(tokens);
+        List<Float> indexesFloat = new ArrayList<>();
+        for (int integer : indexes) {
+            indexesFloat.add((float) integer);
+        }
+        // Preparing the input data
+        NDArray inputs = new NDArray(indexesFloat,
+                new Shape(new int[]{1, inst.seqLength}), context);
+        NDArray tokenTypesND = new NDArray(tokenTypes,
+                new Shape(new int[]{1, inst.seqLength}), context);
+        NDArray validLengthND = new NDArray(new float[] {(float) validLength},
+                new Shape(new int[]{1}), context);
+        List<NDArray> inputBatch = new ArrayList<>();
```

Comment:
You can use Arrays.asList for inputBatch and inputDescs. contexts could use Collections.singletonList